source/wp-content/themes/wporg-developer/inc/advanced-search-filters.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+class Advanced_Search_Filters {
+
+	public static function init() {
+		add_filter( 'the_posts', array( __CLASS__, 'modify_search' ), 10, 2 );
+	}
+
+	/**
+	 * Returns a tax query array for appending wp-parse-source-file taxonomy searches
+	 *
+	 * @param string $name
+	 * @return array
+	 */
+	public static function get_file_tax_query( $name ) {
+		// Get the tax IDs first
+		$tax_ids = get_terms(
+			array(
+				'taxonomy'   => 'wp-parser-source-file',
+				'name__like' => $name,
+				'fields'     => 'ids',
+			)
+		);
+
+		return array(
+			'taxonomy' => 'wp-parser-source-file',
+			'terms'    => $tax_ids,
+		);
+	}
+
+	/**
+	 * Returns a tax query array for appending wp-parser-since taxonomy searches
+	 *
+	 * @param string $term
+	 * @return array
+	 */
+	public static function get_version_tax_query( $term ) {
+		return array(
+			'taxonomy' => 'wp-parser-since',
+			'field'    => 'name',
+			'terms'    => $term,
+		);
+	}
+
+	/**
+	 * Returns a post type string. wp-parser-{post_type}
+	 *
+	 * @param string $post_type_partial
+	 * @return string
+	 */
+	public static function get_post_type_string( $post_type_partial ) {
+		return 'wp-parser-' . strtolower( $post_type_partial );
+	}
+
+	/**
+	 * Modifies the query if user has used advanced search filters
+	 *
+	 * @param \WP_Query $query
+	 * @return void
+	 */
+	public static function modify_query( $query ) {
+		$post_types  = array();
+		$tax_queries = array();
+
+		// Clean up keyword.
+		$keyword = trim( $query->get( 's' ) );
+
+		// Split on spaces.
+		$groups = explode( ' ', $keyword );
+
+		// Loop through groups
+		foreach ( $groups as $group ) {
+			// Check if {qualifier}:{keyword} is used.
+			$split = explode( ':', $group );
+
+			// We have no qualifier.
+			if ( ! isset( $split[1] ) ||
+				empty( $split[0] ) || // Missing a qualifier. Ie: :init
+				$split[1][0] == ':'  // Searching for a class method. Ie: {Class}::init()
+			) {
+
+				// If user has '()' at end of a search string, assume they want a specific function/method.
+				$s = htmlentities( $split[0] );
+				if ( str_contains( $s, '()' ) ) {
+					// Modify the search query to omit the parentheses.
+					$keyword = str_replace( '()', '', $keyword );
```

Comment:
```suggestion
				if ( str_contains( $s, '(' ) ) {
					// Modify the search query to omit the parentheses.
					$keyword = rtrim( $keyword, '()' );
```
This should make it compat with the changes in https://github.com/WordPress/wporg-developer/commit/3c9d9d12a3c5539d7728d648c3fbaa934535a380